compiler/src/iree/compiler/Codegen/SPIRV/SPIRVVectorizeLoadStore.cpp

```diff
@@ -419,6 +436,72 @@ class ProcessInterfaceBindingSubspan final
   }
 };
 
+struct ProcessSubgroupMMALoad final
+    : public MemRefConversionPattern<gpu::SubgroupMmaLoadMatrixOp> {
+  using MemRefConversionPattern::MemRefConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      gpu::SubgroupMmaLoadMatrixOp loadOp, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    auto scalarMemrefType =
+        loadOp.getSrcMemref().getType().dyn_cast<MemRefType>();
+    auto vectorMemrefType =
+        adaptor.getSrcMemref().getType().dyn_cast<MemRefType>();
+
+    if (!vectorMemrefType.getLayout().isIdentity()) {
+      return rewriter.notifyMatchFailure(loadOp,
+                                         "non-identity memref unsupported");
+    }
+
+    Location loc = loadOp.getLoc();
+    auto indices = adjustIndices(scalarMemrefType, vectorMemrefType,
+                                 adaptor.getIndices(), rewriter, loc);
+    if (failed(indices)) return failure();
+
+    auto newLeadDimSize = rewriter.getIntegerAttr(
+        loadOp.getLeadDimensionAttr().getType(),
+        vectorMemrefType.getDimSize(vectorMemrefType.getRank() - 1));
+
+    rewriter.replaceOpWithNewOp<gpu::SubgroupMmaLoadMatrixOp>(
+        loadOp, loadOp.getType(), adaptor.getSrcMemref(), indices.value(),
+        newLeadDimSize);
+    return success();
+  }
+};
+
+struct ProcessSubgroupMMAStore final
+    : public MemRefConversionPattern<gpu::SubgroupMmaStoreMatrixOp> {
+  using MemRefConversionPattern::MemRefConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      gpu::SubgroupMmaStoreMatrixOp storeOp, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    auto scalarMemrefType =
+        storeOp.getDstMemref().getType().dyn_cast<MemRefType>();
+    auto vectorMemrefType =
+        adaptor.getDstMemref().getType().dyn_cast<MemRefType>();
+
+    if (!vectorMemrefType.getLayout().isIdentity()) {
+      return rewriter.notifyMatchFailure(storeOp,
+                                         "non-identity memref unsupported");
+    }
+
+    Location loc = storeOp.getLoc();
+    auto indices = adjustIndices(scalarMemrefType, vectorMemrefType,
+                                 adaptor.getIndices(), rewriter, loc);
+    if (failed(indices)) return failure();
+
+    auto newLeadDimSize = rewriter.getIntegerAttr(
+        storeOp.getLeadDimensionAttr().getType(),
+        vectorMemrefType.getDimSize(vectorMemrefType.getRank() - 1));
```

Comment:
I don't think we should rely on the stride of the memref, we should follow whatever is in the op and just convert based on the type change.

Reply:
Good points! That makes sense. This actually caught a bug. Thanks!